src/Payments.sol

```diff
@@ -119,6 +136,34 @@ contract Payments is
         _;
     }
 
+    modifier onlyRailParticipant(uint256 railId) {
+        require(
+            rails[railId].from == msg.sender ||
+            rails[railId].operator == msg.sender ||
+            rails[railId].to == msg.sender,
+            "failed to authorize: caller is not a rail participant"
+        );
+        _;
+    }
+
+    modifier validateRailNotTerminated(uint256 railId) {
+        require(rails[railId].terminationEpoch == 0, "rail already terminated");
+        _;
+    }
+
+    function terminateOperator(address operator, address token) external {
```

Comment:
I think we can drop this and leave only `setOperatorApproval`.

Reply:
Done.